src/test/java/com/cloudant/tests/util/SimpleHttpServer.java

```diff
@@ -0,0 +1,210 @@
+            } catch (SocketException e) {
+                log.severe("Unable to open server socket");
+                finished.set(true);
+            }
+            // Listening to the port
+            while (!finished.get() && !Thread.currentThread().isInterrupted()) {
+                Socket socket = null;
+                InputStream is = null;
+                OutputStream os = null;
+                try {
+                    log.fine("Server waiting for connections");
+                    //release a permit as we are about to accept connections
+                    semaphore.release();
+                    //block for connections
+                    socket = serverSocket.accept();
+
+                    log.fine("Server accepted connection");
+
+                    is = socket.getInputStream();
+                    os = socket.getOutputStream();
+
+                    //do something with the request and then go round the loop again
+                    serverAction(is, os);
+                } finally {
+                    IOUtils.closeQuietly(is);
+                    IOUtils.closeQuietly(os);
+                    IOUtils.closeQuietly(socket);
+                }
+            }
+            log.fine("Server stopping");
+        } catch (Exception e) {
+            log.log(Level.SEVERE, SimpleHttpServer.class.getName() + " exception", e);
+        } finally {
+            semaphore.release();
+        }
+    }
+
+    /**
+     * Wait up to 2 mintues for the server, longer than that and we give up on the test
+     *
+     * @throws InterruptedException
+     */
+    public void await() throws InterruptedException {
+        semaphore.tryAcquire(2, TimeUnit.MINUTES);
+    }
+
+    /**
+     * @return the url the server is running on, including dynamic port
+     */
+    public String getUrl() {
+        return PROTOCOL + "://127.0.0.1:" + serverSocket.getLocalPort();
+    }
+
+    public SocketAddress getSocketAddress() {
+        return serverSocket.getLocalSocketAddress();
+    }
+
+    /**
+     * Subclasses can override to do something with the input and output streams. The default is
+     * to consume the input stream and write a 200.
+     *
+     * @param is
+     * @param os
+     */
+    protected void serverAction(InputStream is, OutputStream os) throws Exception {
+        readInputLines(is);
+        writeOK(os);
+    }
+
+    /**
+     * Write a simple OK response
+     *
+     * @param os
+     * @throws IOException
+     */
+    protected void writeOK(OutputStream os) throws IOException {
+        BufferedWriter w = new BufferedWriter(new OutputStreamWriter(os));
+        //note the HTTP spec says the status line must be followed by a CRLF
+        //and an empty line must separate headers from optional body
+        //so basically we need 2 CRLFs
+        w.write("HTTP/1.0 200 OK\r\n\r\n");
+        w.flush();
+    }
+
+    /**
+     * Read the input stream lines into the specified List
```

Comment:
There is no list specified in the args. Maybe this should say "Read the input stream lines into {@link #lines}"